File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>hash;

        for(int i=0;i<nums.size();i++){
            int num=nums[i];
            int diff=target-num;
            if(hash.count(diff)!=0){
                return {hash[diff],i};
            }

            hash[num]=i;
        }

        return {-1,-1};
    }
};

File: Data Structures & Algorithms/counting-bits/submission-0.java
class Solution {
    public int hammingWeight(int n) {
        int count=0;

        while(n!=0){
            count+=(n&1);
            n>>>=1;
        }

        return count;
    }
    public int[] countBits(int n) {
        int[] ans=new int[n+1];
        for(int i=0;i<=n;i++){
           int count =hammingWeight(i);
           ans[i]=count;
        }

        return ans;
    }
}
